dss-pades-openpdf/src/main/java/eu/europa/esig/dss/pdf/openpdf/ITextDocumentReader.java

```diff
@@ -102,13 +107,25 @@ public ITextDocumentReader(DSSDocument dssDocument) throws IOException, InvalidP
 	 * @throws IOException if an exception occurs
 	 * @throws InvalidPasswordException if the password is not provided or invalid for a protected document
 	 */
-	public ITextDocumentReader(DSSDocument dssDocument, byte[] passwordProtection) throws IOException, InvalidPasswordException {
+	public ITextDocumentReader(DSSDocument dssDocument, byte[] passwordProtection, ITextPdfMemoryUsageSetting memoryUsageSetting) throws IOException, InvalidPasswordException {
 		Objects.requireNonNull(dssDocument, "The document must be defined!");
 		this.dssDocument = dssDocument;
-		try (InputStream is = dssDocument.openStream()) {
-			this.pdfReader = new PdfReader(is, passwordProtection);
+		try {
+			if (dssDocument instanceof FileDocument) {
+				FileDocument fileDocument = (FileDocument) dssDocument;
+				String filenameSource = fileDocument.getFile().getAbsolutePath();
+				if (ITextPdfMemoryUsageSetting.Mode.FULL.equals(memoryUsageSetting.getMode())) {
+					this.pdfReader = new PdfReader(filenameSource, passwordProtection);
+				} else {
+					this.pdfReader = new PdfReader(new RandomAccessFileOrArray(filenameSource), passwordProtection);
```

Comment:
I reply starting from the last sentences.

There are 2 different but very related topics: loading mode (full or partial) and memory setting (file, memory).
Clearly, they are related because if the loading mode is full than all pdf has to be loaded in memory.

At best of my knowledge, each library offers different level of configuration among such topics.
For OpenPdf, as a conversational reference, take a look at: https://stackoverflow.com/questions/62164906/use-of-randomaccessfileorarray
Check by yourself the code of the PdfReader constructors regards the last loading line: _readPdfPartial();_ for RandomAccessFileOrArray or _readPdf();_ for the other ones.

PdfBox models it differently and inherently uses partial mode when file usage is requested.

----

Regards `new RandomAccessFileOrArray(filenameSource, true, false)` won't be okay because will load all pdf into memory so as a full load (instead of partial load).
Anyway, thanks to your suggestion, I notice the _com.lowagie.text.Document.plainRandomAccess_ attribute and the same semantic into the RandomAccessFileOrArray constructor.
In case of FileDocument AND partial loading I can set that flag to true so the content will be loaded without mapping memory.

I see here an issue of expressivity of the common model and the implementation specific ones. It might be desirable to config whatever combination but I guess that wrapping it is not a solution so just 'reasonable' combination can be provided.

Reply:
Ok, you can leave it for now. I will take a look later in more detail